include/cuco/detail/static_map/static_map_ref.inl

```diff
@@ -231,6 +233,7 @@ static_map_ref<Key, T, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>
     cuco::erased_key<Key>{this->erased_key_sentinel()},
     this->key_eq(),
     this->impl_.probing_scheme(),
+    {},
```

Comment:
`{}` here calls the default ctor of the new `cuda_thread_scope<Scope>` strong type. If not explicitly specified, set/map will use the default `Scope` tparam `cuda::thread_scope_device`. That way we never actually have to deal with the underlying enum values and rely on the libcu++ definitions.

Reply:
I just noticed this is the ctor call in `make_copy`. In this case it will always use the scope of the parent container to initialize the `cuco::cuda_thread_scope` parameter. Let me changed that the way we discussed offline.

Reply:
See #408 and commit [229df4](https://github.com/NVIDIA/cuCollections/pull/408/commits/3229df4ccfa31f4fa1aa94c2b8c8553f61ca7959)

Reply:
I feel there are too many indirection layers to pass the thread scope down to the base implementation. The open addressing base class has a [thread_scope](https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/detail/open_addressing/open_addressing_impl.cuh#L89) defined but the open addressing ref base class doesn't have this definition.

@sleeepyjack Can you please add `thread_scope` to ref base class as well and use it consistently wherever `Scope` is involved? Also, can you please add a dedicated unit test verifying `map_type::thread_scope`, `map_ref_type::thread_scope`, the thread scope of the return ref of `map_ref::make_copy`, etc have the proper thread scope?

Reply:
> Can you please add thread_scope to ref base class

Reply:
> The open addressing base class has a [thread_scope](https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/detail/open_addressing/open_addressing_impl.cuh#L89) defined but the open addressing ref base class doesn't have this definition.

Reply:
OK so this PR is to make CTAD work and #408 is to fix the thread scope issue. Got it.